applications/ShapeOptimizationApplication/custom_utilities/mapping/mapper_vertex_morphing_symmetric.h

```diff
@@ -0,0 +1,524 @@
+// ==============================================================================
+//  KratosShapeOptimizationApplication
+//
+//  License:         BSD License
+//                   license: ShapeOptimizationApplication/license.txt
+//
+//  Main authors:    Armin Geiser, https://github.com/armingeiser
+//
+// ==============================================================================
+
+#ifndef MAPPER_VERTEX_MORPHING_SYMMETRIC_H
+#define MAPPER_VERTEX_MORPHING_SYMMETRIC_H
+
+// ------------------------------------------------------------------------------
+// System includes
+// ------------------------------------------------------------------------------
+#include <iostream>
+#include <string>
+#include <algorithm>
+
+// ------------------------------------------------------------------------------
+// Project includes
+// ------------------------------------------------------------------------------
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "spatial_containers/spatial_containers.h"
+#include "utilities/builtin_timer.h"
+#include "spaces/ublas_space.h"
+#include "mapper_base.h"
+#include "filter_function.h"
+#include "symmetry_base.h"
+#include "symmetry_plane.h"
+#include "symmetry_revolution.h"
+
+// ==============================================================================
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
```

Comment:
Do we still need these empty groups?

Reply:
No (i am never sure if i should keep them or not)

Reply:
I usually don't following @philbucher  :)

Reply:
I usually try to keep the number of lines as little as possible aka remove them

Reply:
I like that approach a lot =) will clean this up.